[Xu Jiheng] iP

JeffZincatz opened the pull request:
# DukePlus

### DukePlus helps you sort out your daily tasks. It is,

* Text-based
* Easy to learn
*  ~~SUPER~~  FAST to use
* Now with simple *GUI*!

### All you need to do is,

1. Download it from [here](https://github.com/nus-cs2103-AY2122S1/ip).
2. Double-click it.
3. Add your tasks.

### Let it manage your tasks for you 😉

### And it is ***FREE***!

## Features:

- [x] Managing tasks
- [x] Save to local data file
- [x] Managing todos, events and deadlines
- [x] Simple GUI (with cute cats!)
- [ ] Reminders (coming soon)

### If you Java programmer, you can use it to practice Java too. Here's the `main` method:

```java
public static void main(String[] args) {
    new Duke("./data/", "duke.txt").run();
}
```

> "The ordinary days that we live in may, in fact, be a series of miracles." -
Koujirou Sasahara

JeffZincatz committed:
abstract Duke greeting message

JeffZincatz committed:
add echo to Duke

JeffZincatz committed:
add indentation to printed messages

JeffZincatz committed:
add add & display list

JeffZincatz committed:
add Task class

JeffZincatz committed:
add ability to mark as done

JeffZincatz committed:
handle some edge cases and exceptions

JeffZincatz committed:
add Todo, Deadline, and Event class

JeffZincatz committed:
Add support for tracking three types of tasks: Todo, Deadline, Event

JeffZincatz committed:
add automated text UI testing

JeffZincatz committed:
fix Duke behaviour and test with automated text UI testing

JeffZincatz committed:
handle errors such as incorrect inputs entered

JeffZincatz committed:
update text ui test for handling errors

JeffZincatz committed:
use an ArrayList<Task> to store the tasks

JeffZincatz committed:
update text ui test for using ArrayList<Task>

JeffZincatz committed:
add support for deleting tasks from the list

JeffZincatz committed:
update text ui testing for deleting tasks

JeffZincatz committed:
abstracted methods for done, todo, event, deadline, and delete commands

JeffZincatz committed:
Add Tasks class which contains tasks io methods

JeffZincatz committed:
Save the tasks automatically whenever the task list changes and load tasks on start up

JeffZincatz committed:
Make the code more OOP with the classes: Ui, Storage, Parser, TaskList, DukeException

JeffZincatz committed:
Use LocalDate as time for Event and Deadline

JeffZincatz committed:
Resolve conflicts

JeffZincatz committed:
Resolve conflicts for merging branch-level-8

JeffZincatz committed:
Organize the classes into suitable java packages

JeffZincatz committed:
Add JUnit tests to test some behaviors of the code

JeffZincatz committed:
Package the App as a JAR file

JeffZincatz committed:
Add more JavaDoc comments

JeffZincatz committed:
Tweak the code to comply with a coding standard

JeffZincatz committed:
Allow to use find to search a task using keywords by description

JeffZincatz committed:
Remove redundant initializers

JeffZincatz committed:
Merge branch 'branch-A-CodingStandard'

JeffZincatz committed:
Resolve conflicts for merging branch-Level-9 with master

JeffZincatz committed:
Tweak header comment mistakes according to coding standard

JeffZincatz committed:
Automate project builds using Gradle

JeffZincatz committed:
Use CheckStyle and tweak code style

JeffZincatz committed:
Automate project builds using Gradle

JeffZincatz committed:
Merge branch 'add-gradle-support'

JeffZincatz committed:
Add a GUI to a simplified Duke with JavaFx

ambroseboo commented on src/main/java/duke/Duke.java:
```diff
@@ -0,0 +1,231 @@
+public class Duke {
+
+    /**
+     * storage instance to handle task list storage.
```
Perhaps you can consider shortening comments for a field declaration to a single line.

ambroseboo commented on src/main/java/duke/task/Deadline.java:
```diff
@@ -0,0 +1,40 @@
+
+public class Deadline extends Task {
+
+    protected LocalDate by;
```
Missing JavaDocs

ambroseboo commented on src/main/java/duke/task/Task.java:
```diff
@@ -0,0 +1,58 @@
+package duke.task;
+
+/**
+ * A task representation for Duke.
```
I feel this definition for Task is ambiguous.

ambroseboo commented on src/test/java/ParserTest.java:
```diff
@@ -0,0 +1,64 @@
+    private static final Parser p = new Parser();
+
+    @Test
+    public void getCommandAction_allCommand_commandStringReturned() {
```
Did you miss out JavaDocs for test methods?

ambroseboo commented on src/main/java/duke/Parser.java:
```diff
@@ -0,0 +1,134 @@
+     *
+     * @param command The command to be parsed.
+     * @return A map of task information from the command.
+     */
```
Needs "throws" field in Javadocs

ambroseboo reviewed (commented):
Not many coding violations at all! 👍

JeffZincatz committed:
Add Duke functionalities to GUI

JeffZincatz committed:
Add header comment JavaDoc for some methods

JeffZincatz committed:
Update README.md to be consistent with PR description

EuniceSim142 commented on src/main/java/duke/Parser.java:
```diff
@@ -0,0 +1,134 @@
+    public Task commandToTask(String command) throws DukeException {
+        switch (getCommandAction(command)) {
+        case "todo":
+            String[] todoSplit = command.split(" ", 2);
```
Collections should be represented in plural, so maybe change to todoDetails instead.
```suggestion
            String[] todoDetails = command.split(" ", 2);
```

EuniceSim142 commented on src/main/java/duke/Parser.java:
```diff
@@ -0,0 +1,134 @@
+            }
+            return new Todo(todoDescription);
+        case "event":
+            String[] eventSplit = command.split(" /at ");
```
(same as todo)
Collections should be represented in plural, so maybe change to eventDetails instead.
```suggestion
            String[] eventDetails = command.split(" /at ");
```

EuniceSim142 commented on src/main/java/duke/Storage.java:
```diff
@@ -0,0 +1,137 @@
+    }
+
+    /**
+     * Read data from the data file duke.txt.
```
Method header comments should start with "Reads" instead of "Read"

EuniceSim142 commented on src/main/java/duke/task/TaskList.java:
```diff
@@ -0,0 +1,150 @@
+     * @return A string of tasks matching the patter.
+     */
+    public String getMatchedTasksString(String pattern) {
+        ArrayList<Task> matched = new ArrayList<>();
```
plural representation for collections
```suggestion
        ArrayList<Task> matches = new ArrayList<>();
```

EuniceSim142 reviewed (commented):
Looks Good! Very little coding standard violations. 
Although maybe take note that Javadocs method header comments should start like Returns..., Gets... or Prints..., instead of Return..., Get... or Print...

JeffZincatz committed:
Assertions: add relevant assert statements in methods

Assertions is needed to check for potential errors in methods.

Some assertions help locate errors where assuptions are not fulfil.

Let's,
Add some relevant assert statements in methods to test for potential
assumption mismatches. For example, check that for each type of command,
the index used to access the descriptions and time (if applicable) is
not out of bound.

JeffZincatz committed:
Improve code quality

JeffZincatz committed:
Merge pull request #1 from JeffZincatz/branch-A-Assertions

Assertions: add relevant assert statements in methods

JeffZincatz committed:
Merge pull request #2 from JeffZincatz/branch-A-CodeQuality

Improve code quality

JeffZincatz committed:
Sort tasks of type Deadline and Event by time

JeffZincatz committed:
Merge pull request #3 from JeffZincatz/branch-C-Sort

Sort tasks of type Deadline and Event by time

JeffZincatz committed:
Exit app properly with bye command

JeffZincatz committed:
Add greeting message at the start

JeffZincatz committed:
Rename Duke to MemoCat

JeffZincatz committed:
Refactor README.md as MemoCat

JeffZincatz committed:
Add a representative screenshot of MemoCat

JeffZincatz committed:
Add a User Guide

JeffZincatz committed:
Set theme jekyll-theme-cayman

JeffZincatz committed:
Code Quality: Refactors method header comments

JeffZincatz committed:
SLAP: Reduces methods length

JeffZincatz committed:
Merge branch 'master' of https://github.com/JeffZincatz/ip
